stac_geoparquet/json_reader.py

```diff
@@ -0,0 +1,28 @@
+"""Return an iterator of items from an ndjson, a json array of items, or a featurecollection of items."""
+
+import orjson
+from typing import Iterator, Dict, Any, Union, Iterable
+from pathlib import Path
+
+
+def read_json(
+    path: Union[str, Path, Iterable[Union[str, Path]]]
+) -> Iterator[Dict[str, Any]]:
+    """Read a json or ndjson file."""
+    if isinstance(path, (str, Path)):
+        path = [path]
+
+    for p in path:
+        with open(p) as f:
+            try:
+                # read ndjson
+                for line in f:
+                    yield orjson.loads(line.strip())
```

Comment:
Couldn't there be a JSON file (not a newline-delimited json file) without newlines where this would fail? Or at least where this would yield a list of features at once instead of iterating over them?

```py
import json
import orjson

l = [{"a": 1}, {"a": 2}]
with open('tmp.json', 'w') as f:
    json.dump(l, f, separators=(",", ":"))

def read():
    with open("tmp.json") as f:
        for line in f:
            yield orjson.loads(line.strip())

out = list(read())
len(out)
# 1
```

I'm really not a fan of guessing whether the user has JSON or newline-delimited JSON data. Up until now I think we've assumed that they've always had newline-delimited json. If we need to support JSON as well, can we create a different function for this?

Reply:
note to self:
1. if it's a list, yield from that list
2. if it's a dict with a `"features"` key, yield from the list of features,
3. otherwise yield that item